### ISSUE TITLE
Restarting failed builds on Azure spawns 2 extra runs

### ISSUE THREAD

**Issue by carterbox** [conda-forge/conda-forge.github.io#2016]
### Conda-forge documentation

- [X] I could not solve my problem using the conda-forge documentation.

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

### Issue

In the libmagma-feedstock, builds on the main branch often time-out because the builds take 5.5 +/- 0.5 hours to complete. The build time is quite variable; I don't know why and that's not the purpose of this issue.

The issue is that when I return and click the "Rerun failed jobs" button via the GitHub UI. Two extra runs are spawned. For example, after merging the recent update to version 2.7.2. The post-merge commit had two failed build variants, so I pushed the re-run failed jobs button.

![Screenshot from 2023-10-12 12-42-00](https://github.com/conda-forge/conda-forge.github.io/assets/9604511/37b00f9f-e1d8-464b-b750-ef2adb5d03e1)

https://dev.azure.com/conda-forge/feedstock-builds/_build?definitionId=18893&_a=summary

However, the result is that runs 20231012.2 and 20231012.3 are spawned (which is not necessary?). These later two runs are re-running all of the build variants, whereas run 20231012.1 is only rerunning the two failed variants (which is what I requested).

**Comment by hmaarrfk** [comment]
I've noticed this too. you want to wait until all the jobs come to completion, fail or pass.

**Comment by carterbox** [comment]
I think I did wait until all the jobs were completed? I'll keep that in mind for next time.

**Comment by carterbox** [comment]
Yes. This was after all jobs had completed. 2 timed out and the rest passed.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=802513&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&s=96ac2280-8cb4-5df5-99de-dd2da759617d

**Comment by h-vetinari** [comment]
It doesn't just affect libmagma (though I've [observed](https://github.com/conda-forge/libmagma-feedstock/pull/9#issuecomment-1758574833) this on another PR there very recently). Here's two more recent examples, the most extreme of which spawned >20(!) jobs.

* [petsc](https://github.com/conda-forge/petsc-feedstock/pull/170#issuecomment-1744393215)
* [pandas](https://github.com/conda-forge/pandas-feedstock/pull/183#issuecomment-1742241852)

In the past, my understanding was that this was due to people clicking the "re-run" button in the Github UI several times, or before the CI run on Azure had fully completed, but it seems to happen as well under textbook conditions (i.e. wait for AP to finish, then click "re-run" once).
